docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,487 @@
+===============================================================================
+
+One of the primary benefits of asynchronous operations is the ability to
+overlap data transfer with kernel execution, leading to better resource
+utilization and improved performance.
+
+Asynchronous execution is particularly advantageous in iterative processes. For
+instance, if an iteration calculation is initiated, it can be efficient to
+prepare the input data simultaneously, provided that this preparation does not
+depend on the kernel's execution.
+
+Querying device capabilities
+-------------------------------------------------------------------------------
+
+Some AMD HIP-enabled devices can perform asynchronous memory copy operations to
+or from the GPU concurrently with kernel execution. Applications can query this
+capability by checking the ``asyncEngineCount`` device property. Devices with
+an ``asyncEngineCount`` greater than zero support concurrent data transfers.
+Additionally, if host memory is involved in the copy, it should be page-locked
+to ensure optimal performance.
+
+Asynchronous memory operations
+-------------------------------------------------------------------------------
+
+Asynchronous memory operations allow data to be transferred between the host
+and device while kernels are being executed on the GPU. Using operations like
+:cpp:func:`hipMemcpyAsync`, developers can initiate data transfers without
+waiting for the previous operation to complete. This overlap of computation and
+data transfer ensures that the GPU is not idle while waiting for data. Examples
+include launching kernels in one stream while performing data transfers in
+another. This technique is especially useful in applications with large data
+sets that need to be processed quickly.
+
+Concurrent data transfers
+-------------------------------------------------------------------------------
+
+Concurrent data transfers are supported between the host and device, within
+device memory, and among multiple devices. Using :cpp:func:`hipMemcpyAsync`,
+data can be transferred asynchronously, allowing for efficient data movement
+without blocking other operations. :cpp:func:`hipMemcpyPeerAsync` enables data
+transfers between different GPUs, facilitating multi-GPU communication.
+Concurrent data transfers are important for applications that require frequent
+and large data movements. By overlapping data transfers with computation,
+developers can minimize idle times and enhance performance. This is
+particularly important for applications that need to handle large volumes of
+data efficiently.
+
+Concurrent data transfers with intra-device copies
+-------------------------------------------------------------------------------
+
+It is also possible to perform intra-device copies simultaneously with kernel
+execution on devices that support the ``concurrentKernels`` device property
+and/or with copies to or from the device (for devices that support the
```

Comment:
```suggestion
and/or with copies to or from the device (for devices that support the
```
Are copies to or from the device intra-device copies?

Reply:
No, intra-device copying is data transfer within the same device.

Thank you for drawing my attention to it. The text was misleading, so I rephrased it.